superset/sql_parse.py

```diff
@@ -0,0 +1,89 @@
+import sqlparse
+from sqlparse.sql import IdentifierList, Identifier
+from sqlparse.tokens import Keyword, Name
+
+
+RESULT_OPERATIONS = {'UNION', 'INTERSECT', 'EXCEPT'}
+PRECEDES_TABLE_NAME = {'FROM', 'JOIN', 'DESC', 'DESCRIBE', 'WITH'}
+
+
+def precedes_table_name(token_value):
+    for keyword in PRECEDES_TABLE_NAME:
+        if keyword in token_value:
+            return True
+    return False
+
+
+def get_full_name(identifier):
+    if len(identifier.tokens) > 1 and identifier.tokens[1].value == '.':
+        return "{}.{}".format(identifier.tokens[0].value,
+                              identifier.tokens[2].value)
+    return identifier.get_real_name()
+
+
+def is_result_operation(keyword):
+    for operation in RESULT_OPERATIONS:
+        if operation in keyword.upper():
+            return True
+    return False
+
+
+def is_identifier(token):
+    return isinstance(token, IdentifierList) or isinstance(token, Identifier)
+
+
+def process_identifier(identifier, table_names, aliases):
+    # exclude subselects
+    if '(' not in '{}'.format(identifier):
+        table_names.append(get_full_name(identifier))
```

Comment:
yeah, it's more c++ approach, I'll try to rewrite it using yield - it will be more pythonic way